File: Classes/EntityClasses/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MoOnlineStore.Core.EntityClasses.OrderAggregate
{
   public class Order:BaseEntity
    {
        public Order()
        {
        }

        public Order(IReadOnlyList<OrderItem> orderItem, string buyerEmail, Address eliveryAddress, decimal subtotal, DeliveryMethod deliveryMethod,string paymentIntendID)
        {
            OrderItems = orderItem;
            BuyerEmail = buyerEmail ;
            this.ShipToAddress = eliveryAddress ;
            Subtotal = subtotal;
            this.DeliveryMethod = deliveryMethod ;
            this.PaymentIntentId = paymentIntendID;
        }

        public string BuyerEmail { get; set; }
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
        public Address ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }
    }
}

File: client/src/app/shared/Models/Order.ts
import { IAddress } from './IAddress';
export interface IOrderToCreate {
  basketId: string;
  deliveryMethodId: number;
  shipToAddress: IAddress;
}
export interface Iorder {
  id: number;
  buyerEmail: string;
  orderDate: string;
  shipToAddress: IAddress;
  deliveryMethod: string;
  shippingPrice: number;
  orderItems: OrderItem[];
  subtotal: number;
  total: number;
  status: string;
}

export interface OrderItem {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}
